app.py

```diff
@@ -0,0 +1,46 @@
+import io
+from flask import Flask, request
+import pyarrow as pa
+import pyarrow.ipc as ipc
+
+
+app = Flask(__name__)
+
+
+# Dummy dataset for now
+def make_reader():
+    data = pa.table([range(26), "abcdefghijklmnopqrstuvwxyz"], ["int", "string"])
+    return pa.RecordBatchReader.from_batches(data.schema, data.to_batches())
+
+
+# iterator of bytes (how Flask streams output)
+def make_result_generator(reader, options):
+    with io.BytesIO() as f, ipc.new_stream(f, reader.schema, options=options) as stream:
+        yield f.getvalue()
+
+        for batch in reader:
+            f.seek(0)
+            f.truncate(0)
+            stream.write_batch(batch)
+            yield f.getvalue()
```

Comment:
We may need an explicit `close()`:

```suggestion
            yield f.getvalue()

        f.seek(0)
        f.truncate(0)
        stream.close()
        yield f.getvalue()
```

Reply:
> I think the stream closing is handled by the context manager?

Yes. I think so too. But I think that we need to `yield` a data written by `stream.close()` (EOS (end-of-stream) will be written by `stream.close()`).
See also: https://arrow.apache.org/docs/format/Columnar.html#ipc-streaming-format

If `app.response_class()` returns the (optional) EOS to a client without this, we don't need this.